bin/src/learner/predict.py

```diff
@@ -0,0 +1,67 @@
+import torch
+from torch.utils.data import DataLoader
+from ..data.handlertorch import TorchDataset
+from ..utils.performance import Performance
+
+class PredictWrapper():
+    def __init__(self, model, data_path: str, experiment: object, loss_dict: dict, split: int, batch_size: int):
+        self.model = model
+        self.loss_dict = loss_dict
+        self.dataloader = DataLoader(TorchDataset(data_path, experiment, split=split), batch_size=batch_size, shuffle=False)
+
+    def predict(self) -> dict:
+        """
+        Predicts the output of the model on the data.
+        """
+        self.model.eval()
+        predictions = {k:[] for k in list(self.dataloader)[0][1].keys()}   # list(self.dataloader)[0] is the first batch of dataloader, then the element [1] is y.
+        with torch.no_grad():
```

Comment:
write a comment explaining a bit more what is happening here. many for loops and it is hard to keep track of what is happening.